### PR TITLE
Feature #180537199 – Import proteomics differential experiments

### PR DESCRIPTION
**Pull request by alfonsomunozpomer**
Add validation of proteomics differential experiments to be like differential RNA-Seq, but without raw counts files. It includes some `atlas-web-core` changes (see below).

This PR is accompanied by https://github.com/ebi-gene-expression-group/atlas-web-core/pull/86.

### CODE REVIEW

File: app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExperimentCheckerIT.java [hunk 1]
```diff
@@ -66,6 +68,8 @@ void checkAllFiles(String experimentAccession, ExperimentType experimentType) {
         return Stream.of(
                 RNASEQ_MRNA_BASELINE,
                 PROTEOMICS_BASELINE,
+                PROTEOMICS_DIFFERENTIAL,
+                PROTEOMICS_BASELINE_DIA,
```
Comment (upendrakumbham) [inline app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExperimentCheckerIT.java:72]: Thanks for Adding 'DIA' experiment type here

Comment (upendrakumbham) [reply app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExperimentCheckerIT.java:72]: I ran this class, one of the test cases is failing.
Here is the details: 
```
ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[2] FAILED
    java.lang.IllegalStateException at ExperimentCheckerIT.java:64
```
Can you please run this class on your side?

File: app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExpressionAtlasExperimentCheckerTest.java [hunk 2]
```diff
@@ -160,7 +155,9 @@ private void fileMatchesHeaderBasedOnAssayGroups(BiConsumer<String, List<String[
                 addFile);
 
         assertFails(
-                "<assay_group id=\"g1\"><assay>A</assay></assay_group>",
+                "<assay_group id=\"g1\">" +
+                        "<assay>A</assay>" +
```
Comment (upendrakumbham) [inline app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExpressionAtlasExperimentCheckerTest.java:159]: All `ExpressionAtlasExperimentCheckerTest.java` tests passed.

**Review by upendrakumbham — CHANGES_REQUESTED** [review]
The code looks good to me and has lots of cleanup.

I ran this class, one of the test case is failing.  
```
Here is the details:
ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[2] FAILED
    java.lang.IllegalStateException at ExperimentCheckerIT.java:64

```
Can you please run this class on your side?

I'm not sure why cli and war checks are failing in the PR, can you please check.

File: app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExperimentCheckerIT.java (hunk 1 above)
Comment (upendrakumbham) [reply app/src/test/java/uk/ac/ebi/atlas/experimentimport/ExperimentCheckerIT.java:72]: I think we should think about renaming this `ExperimentCheckerIT.java` to `ExpressionAtlasExperimentCheckerIT.java` as ExperimentChecker is an interface.

**Comment by alfonsomunozpomer** [comment]
I run all the tests and they passed:
```bash
ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[1] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[2] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[3] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[4] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[5] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[6] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[7] PASSED

ExperimentCheckerIT > checkAllFiles(String, ExperimentType) > uk.ac.ebi.atlas.experimentimport.ExperimentCheckerIT.checkAllFiles(String, ExperimentType)[8] PASSED

BUILD SUCCESSFUL in 10s
7 actionable tasks: 1 executed, 6 up-to-date
```

Maybe you need to add E-PROT-39 to your integration environment (Postgres and Solr)?

**Comment by alfonsomunozpomer** [comment]
Probably the build fails because I was too hasty and deleted the remote branch in `atlas-web-core`, let me restore it and I’ll try again.

**Comment by upendrakumbham** [comment]
> Probably the build fails because I was too hasty and deleted the remote branch in `atlas-web-core`, let me restore it and I’ll try again.

Yeah, @alfonsomunozpomer. I don't have E-PROT-39 in the Solr.

**Review by upendrakumbham — APPROVED** [review]
All LGTM! Please merge away.

**Comment by pcm32** [comment]
This PR broke the Github actions on 14e4d53 I suspect...

**Comment by pcm32** [comment]
Here is the message https://github.com/ebi-gene-expression-group/atlas-web-bulk/actions/runs/1718175139 (same for both I think).

**Pull request merged**